endpoints/data_objects/src/main.cpp

```diff
@@ -850,7 +850,53 @@ namespace
 					logging::trace(
 						*_sess_ptr, "{}: Opening stream for reading to data object [{}].", fn, lpath_iter->second);
 					auto tp = std::make_unique<io::client::native_transport>(dedicated_conn);
-					io::idstream in{*tp, lpath_iter->second};
+					io::idstream in;
+
+					if (auto iter = _args.find("resource"); iter != std::end(_args)) {
+						if (_args.find("replica-number") != std::end(_args)) {
+							logging::error(
+								*_sess_ptr, "{}: [resource] and [replica-number] parameters are incompatible.", fn);
+							res.result(http::status::bad_request);
+							res.prepare_payload();
+							return _sess_ptr->send(std::move(res));
+						}
+
+						logging::debug(
+							*_sess_ptr,
+							"{}: Opening replica of [{}] on resource [{}].",
+							fn,
+							lpath_iter->second,
+							iter->second);
+						in.open(*tp, lpath_iter->second, io::root_resource_name{iter->second});
+					}
+					else if (iter = _args.find("replica-number"); iter != std::end(_args)) {
+						if (_args.find("resource") != std::end(_args)) {
+							logging::error(
+								*_sess_ptr, "{}: [resource] and [replica-number] parameters are incompatible.", fn);
+							res.result(http::status::bad_request);
+							res.prepare_payload();
+							return _sess_ptr->send(std::move(res));
+						}
+
+						int value = -1;
+						try {
+							value = std::stoi(iter->second);
```

Comment:
Just checking: We're leaning on the server to catch invalid replica numbers (i.e. negative values)?

Reply:
After some discussion, we've agreed on the following:
- The HTTP API is a passthrough implementation
- The HTTP API will NOT check for incompatible parameters
- The HTTP API can check type information, but should NOT duplicate iRODS server functionality

These rules should be documented in the repo and help keep the HTTP API future-proof with changes in iRODS server behavior.

Reply:
> Just checking: We're leaning on the server to catch invalid replica numbers (i.e. negative values)?

Yes. We're leaning on the iRODS server to catch invalid replica numbers.

> After some discussion, we've agreed on the following:
>
> - The HTTP API is a passthrough implementation
> - The HTTP API will NOT check for incompatible parameters
> - The HTTP API can check type information, but should NOT duplicate iRODS server functionality
>
> These rules should be documented in the repo and help keep the HTTP API future-proof with changes in iRODS server behavior.

Created #456.